code/modules/security/genpop.dm

```diff
@@ -0,0 +1,650 @@
+//From NSV13
+//Credit to oraclestation for the idea! This just a recode...
+// Recode CanAllowThrough() and machine_stat
+
+/obj/machinery/turnstile
+	name = "turnstile"
+	desc = "A mechanical door that permits one-way access to the brig."
+	icon = 'icons/obj/machines/turnstile.dmi' //ADD ICON
+	icon_state = "turnstile_map" //ADD ICON
+	power_channel = AREA_USAGE_ENVIRON
+	density = TRUE
+	pass_flags_self = PASSTRANSPARENT | PASSGRILLE | PASSSTRUCTURE
+	obj_integrity = 250
+	max_integrity = 250
+	integrity_failure = 74
+	//Robust! It'll be tough to break...
+	armor = list("melee" = 50, "bullet" = 20, "laser" = 0, "energy" = 80, "bomb" = 10, "bio" = 100, "rad" = 100, "fire" = 90, "acid" = 50, "stamina" = 0)
+	anchored = TRUE
+	idle_power_usage = 2
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	layer = OPEN_DOOR_LAYER
+	//Seccies and brig phys may always pass, either way.
+	req_one_access = list(ACCESS_BRIG, ACCESS_BRIGPHYS, ACCESS_PRISONER)
+	//Cooldown so we don't shock a million times a second
+	COOLDOWN_DECLARE(shock_cooldown)
+	circuit = /obj/item/circuitboard/machine/turnstile
+	var/state = TURNSTILE_SECURED
+
+/obj/item/circuitboard/machine/turnstile
+	name = "Turnstile circuitboard"
+	desc = "The circuit board for a turnstile machine."
+	build_path = /obj/machinery/turnstile
+	req_components = list(
+		/obj/item/stack/cable_coil = 5,
+		/obj/item/stock_parts/micro_laser = 2,
+		/obj/item/stack/ore/bluespace_crystal = 1,
+		/obj/item/stack/rods = 12
+	)
+
+/obj/machinery/turnstile/examine(mob/user)
+	. = ..()
+	if(state == TURNSTILE_SECURED)
+		. += "<span class='notice'>The turnstile panel is tightly <b>screwed</b> to the frame.</span>"
+	if(state == TURNSTILE_CIRCUIT_EXPOSED)
+		. += "<span class='notice'>The turnstile circuitboard is exposed, you could <b>pry it</b> from the frame.</span>"
+	if(state == TURNSTILE_SHELL)
+		. += "<span class='notice'>The turnstile frame is empty, ready to be sliced apart through <b>welding</b>.</span>"
+
+//Executive officer's line variant. For rule of cool.
+/*/obj/machinery/turnstile/xo
+	name = "\improper XO line turnstile"
+	req_one_access = list(ACCESS_BRIG, ACCESS_HEADS)
+*/
+
+/obj/structure/closet/secure_closet/genpop
+	name = "Prisoner locker"
+	desc = "A locker to store a prisoner's valuables, that they can collect at a later date."
+	req_access = list(ACCESS_BRIG)
+	anchored = TRUE
+	locked = FALSE
+	var/registered_name = null
+
+/obj/structure/closet/secure_closet/genpop/Initialize(mapload)
+	. = ..()
+
+/obj/structure/closet/secure_closet/genpop/attackby(obj/item/W, mob/user, params)
+	var/obj/item/card/id/I = null
+	if(istype(W, /obj/item/card/id))
+		I = W
+	else
+		I = W.GetID()
```

Comment:
```suggestion
	var/obj/item/card/id/I = W.GetID()
```
If the item is an ID, then calling GetID on it will return itself.
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/a6d6c845-882d-4cec-a566-81a2fee7977c)

Reply:
If W is an ID then I = W, else I = W.GetID()
That said, I'm not sure what this check is for, hands I guess? It's inherited from NSV and I don't think we need to check for that here?